pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -526,6 +538,9 @@ private File downloadSegmentFromDeepStore(String segmentName, SegmentZKMetadata
     } else {
       try {
         File tarFile = downloadAndDecrypt(segmentName, zkMetadata, tempRootDir);
+        if (_isRetrySegmentDownloadUntarFailure) {
```

Comment:
Here we want to retry on download + untar together. Why not putting the retry logic over the `downloadSegmentFromDeepStore()`?

Reply:
Apart from retry changes in this PR, downloadSegmentFromDeepStore() has only one other flow downloadAndStreamUntarWithRateLimit() in which a retry logic is already present for that path in HttpSegmentfetcher.fetchUntarSegmentToLocalStreamed() for download failure only. Should we handle untar failures also there itself ? As both download and untar happens in that same function 

Te advantage of: `retry logic over the downloadSegmentFromDeepStore()` is that it required less code changes.

Reply:
The same retry logic also exists in `downloadAndDecrypt()`. I feel the clean fix should be change the `downloadAndDecrypt()` to `downloadAndDecryptUntar()` which also untar the segment similar to `downloadAndStreamUntarWithRateLimit()`. We already have the configurable retries within the `SegmentFetcher`, and IMO we don't want another set of configs to control the retry for untar.
cc @jasperjiaguo

Reply:
Agree on this, let me refactor and update the PR.

Reply:
+1. I also think `downloadAndDecryptUntar` would be cleaner